mod_wasm/modules/wasm/mod_wasm.c

```diff
@@ -204,67 +167,31 @@ static void *create_server_config(apr_pool_t *p, server_rec *s)
     cfg = (x_cfg *) apr_pcalloc(p, sizeof(x_cfg));
     cfg->local = 0;
     cfg->congenital = 0;
-    cfg->bEnableCGI = 0;
+    cfg->bWasmEnableCGI = 0;
     cfg->cmode = CONFIG_MODE_SERVER;
     /*
      * Note that we were called in the trace list.
      */
     sname = (sname != NULL) ? sname : "";
     cfg->loc = apr_pstrcat(p, "SVR(", sname, ")", NULL);
     trace_nocontext(NULL, __FILE__, __LINE__, sname);
-    return (void *) cfg;
-}
-
-/*
- * Post-config hook
- */
-static int post_config_hook(apr_pool_t *pconf, apr_pool_t *plog,
-                          apr_pool_t *ptemp, server_rec *s)
-{
-    trace_nocontext(NULL, __FILE__, __LINE__, "post_config_hook() - initializing wasm_runtime...");
-
-    // init wasm_runtime
-    const char* result = wasm_runtime_init_module();
-
-    if (strcmp(result, "") != 0)  // something went wrong
-        trace_nocontext(NULL, __FILE__, __LINE__, "post_config_hook() - ERROR! Couldn't initiale wasm_runtime!");
-    else
-        trace_nocontext(NULL, __FILE__, __LINE__, "post_config_hook() - wasm_runtime initialized!");
 
-    return_const_char_ownership(result);
+    // creates a new Wasm config for the current context
+    wasm_config_create(cfg->loc);
 
-    return OK;
+    return (void *) cfg;
 }
 
 // Add the provided key to the wasmtime runtime as an environment
 // variable.
-static int _wasm_config_add_env(void *h_, const char *key, const char *value)
+static int _wasm_executionctx_env_add(void* context, const char *key, const char *value)
 {
-    wasm_config_add_env(key, value);
-    return 1;
-}
-
-// Populates the runtime with the arguments defined in the static
-// configuration.
-static void populate_runtime_with_config_defined_args(x_cfg *cfg)
-{
-    for (int i = 0; i < cfg->configArgCount; ++i) {
-        wasm_config_add_arg(cfg->configArgs[i]->arg);
-    }
-}
+    if ( wasm_executionctx_env_add( (const char*)context, key, value) != OK )
+        trace_nocontext(NULL, __FILE__, __LINE__, "_wasm_executionctx_env_add() - ERROR! Couldn't add env variable to Wasm execution context!");
 
-// Populates the runtime with the environment variables defined in the
-// static configuration.
-static void populate_runtime_with_config_defined_envs(x_cfg *cfg)
-{
-    for (int i = 0; i < cfg->configEnvVarCount; ++i) {
-        configEnvVar *envVar = cfg->configEnvVars[i];
-        wasm_config_add_env(envVar->key, envVar->value);
-    }
+    return 1;
```

Comment:
Out of curiosity, what does this `1` mean? How does it interact with `apr_table_do`? I assume we could also return NULL, or am I misunderstanding?

Reply:
From `apr_tables.h` when defining `apr_table_do()`:
>  If an invocation of the @p comp function returns zero,
>  iteration will continue using the next specified key, if any.

Since we are invoking the function with the final `vargs` key parameter as `NULL`, that return value doesn't really matter.

```c
      apr_table_do(_wasm_executionctx_env_add, (void*)exec_ctx_id, r->subprocess_env, NULL);
```